riscv/src/elf.rs

```diff
@@ -0,0 +1,1213 @@
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    collections::{btree_map::Entry, BTreeMap, BTreeSet, HashMap},
+    fs,
+    path::Path,
+};
+
+use goblin::{
+    elf::sym::STT_OBJECT,
+    elf::{
+        header::{EI_CLASS, EI_DATA, ELFCLASS32, ELFDATA2LSB, EM_RISCV, ET_DYN},
+        program_header,
+        reloc::{R_RISCV_32, R_RISCV_HI20, R_RISCV_RELATIVE},
+        sym::STT_FUNC,
+        Elf,
+    },
+};
+use itertools::{Either, Itertools};
+use powdr_asm_utils::data_storage::SingleDataValue;
+use powdr_number::FieldElement;
+use raki::{
+    decode::Decode,
+    instruction::{Extensions, Instruction as Ins, OpcodeKind as Op},
+    Isa,
+};
+
+use crate::{
+    code_gen::{self, InstructionArgs, MemEntry, Register, RiscVProgram, Statement},
+    Runtime,
+};
+
+struct ElfProgram {
+    symbol_table: SymbolTable,
+    data_map: BTreeMap<u32, Data>,
+    text_labels: BTreeSet<u32>,
+    instructions: Vec<HighLevelInsn>,
+    entry_point: u32,
+}
+
+pub fn elf_translate<F: FieldElement>(
```

Comment:
What does it translate to? The return type is `String`, so it's a lot of guessing :)

Reply:
This return the powdr asm contents.